numpy/distutils/intelccompiler.py

```diff
@@ -15,14 +17,15 @@ class IntelCCompiler(UnixCCompiler):
 
     def __init__(self, verbose=0, dry_run=0, force=0):
         UnixCCompiler.__init__(self, verbose, dry_run, force)
-        self.cc_exe = 'icc -fPIC'
+        self.cc_exe = ('icc -fPIC -fp-model strict -O3 '
+                       '-fomit-frame-pointer -openmp')
         compiler = self.cc_exe
         self.set_executables(compiler=compiler,
                              compiler_so=compiler,
                              compiler_cxx=compiler,
                              archiver='xiar' + ' cru',
-                             linker_exe=compiler,
-                             linker_so=compiler + ' -shared')
+                             linker_exe=compiler + ' -shared-intel',
+                             linker_so=compiler + ' -shared -shared-intel')
```

Comment:
Are you sure? it will lead to dependency on shared libraries from Intel so that resulting application become less portable.